lib/game.rb

```diff
@@ -0,0 +1,74 @@
+class Game
+    attr_accessor :board, :player_1, :player_2
+
+    WIN_COMBINATIONS = [
+        [0,1,2],
+        [3,4,5],
+        [6,7,8],
+        [0,3,6],
+        [1,4,7],
+        [2,5,8],
+        [0,4,8],
+        [6,4,2]
+    ]
+
+    def initialize(player_1 = nil, player_2 = nil, board = nil)
+        player_1 = Players::Human.new("X") if player_1.nil?
+        player_2 = Players::Human.new("O") if player_2.nil?
+        board.nil? ? @board = Board.new : @board = board
+        @player_1 = player_1
+        @player_2 = player_2
+    end
+
+    def self.combinations
+        WIN_COMBINATIONS
+    end
+
+    def current_player
+        self.board.turn_count.even? ? self.player_1 : self.player_2
+    end
+
+    def over?
+        self.draw? || self.won?
+    end
+
+    def won?
+        WIN_COMBINATIONS.any? {|array| array.all? {|cell| self.board.cells[cell] == "X"}} ||WIN_COMBINATIONS.any? {|array| array.all? {|cell| self.board.cells[cell] == "O"}}
+    end
+
+    def draw?
+        self.board.full? && !self.won?
+    end
+
+    def winner
```

Comment:
You can take advantage of the `won?` method here:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.